src/components/ScreenStack.tsx

```diff
@@ -19,8 +23,14 @@ function ScreenStack(props: ScreenStackProps) {
     const isFreezeEnabled =
       descriptor?.options?.freezeOnBlur ?? freezeEnabled();
 
+    // On Fabric, when screen is frozen, animated and reanimated values are not updated
+    // due to component being unmounted. To avoid this, we don't freeze the previous screen there
+    const freezePreviousScreen = isFabric()
+      ? size - index > 2
+      : size - index > 1;
```

Comment:
We need to keep in mind that this is only a heuristics anyway and animations won't work on third-top screen etc. 

For now we can proceed.

Reply:
I can only see this as a problem when we have multiple transparent modals, in which case I wouldn't enable freeze on that navigator anyway.